From a6e59f54dee611709842f4cce2059e0ecace5c03 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Mon, 9 Sep 2024 14:39:59 +0200
Subject: [PATCH 1076/1695] dts: stm32: describe vrefbuf supply for stm32mp25

Describe PWR VDDA18ADC as VREFBUF supply for STM32MP25.
The goal is to guaranty that VDDA18ADC SV is set before enabling VREFBUF.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git dbeb7fbc94526520ca2f5d01e3856267b138e0fe dts: stm32: describe vrefbuf supply for stm32mp25]
Change-Id: Idc6da470f816372cd509e0124cadc25954ade61e
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/403072
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Tested-by: Pascal PAILLET <p.paillet@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 3e4847604..6c6558127 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -810,6 +810,7 @@
 			regulator-min-microvolt = <1210000>;
 			regulator-max-microvolt = <1500000>;
 			clocks = <&rcc CK_BUS_VREF>;
+			vdda-supply = <&vdda18adc>;
 			status = "disabled";
 		};
 
-- 
2.47.2

